tests/LinkDotNet.Blog.UnitTests/Domain/BlogPostTests.cs

```diff
@@ -24,6 +24,21 @@ public void ShouldUpdateBlogPost()
         blogPostToUpdate.PreviewImageUrlFallback.Should().Be("Url2");
         blogPostToUpdate.IsPublished.Should().BeTrue();
         blogPostToUpdate.Tags.Should().BeNullOrEmpty();
+        blogPostToUpdate.Slug.Should().NotBeNull();
+    }
+
+    [Theory]
+    [InlineData("blog title","blog-title")]
+    [InlineData("blog      title", "blog-title")]
+    [InlineData("blog +title", "blog-title")]
+    [InlineData("blog/title", "blogtitle")]
+    [InlineData("blog /title", "blog-title")]
```

Comment:
I would add some more cases like "blog-title", "blog.title" and more cases on top of our mind.

We should also think of a "max-length" for the slug. Not sure how other platforms handle that.

Reply:
Here are some more test cases we need to cover:
- [ ] `BLOG TITLE` -> `blog-title`
- [ ] `àccent` -> `accent`
- [ ] `get 100$ quick` -> `get-100-quick`
- [ ] `blog,title` -> `blog-title`
- [ ] `use,comma` -> `use-comma`
- [ ] `blog?!title` -> `blogtitle`
- [ ] `blog----title` -> `blog-title`

Reply:
I am thinking of some kind of algorithm for handling max-length in slug.
we can split blog titles by space and append them together until the slug doesn't break max-length rules by doing this not only can we limit the slug to obey max-length rules but also we can avoid crazy slugs.

For instance, our blog title is `How to set up a firewall with ufw on Ubuntu` and our max-length is 27 char, the slug is going to be something like this `how-to-set-up-a-firewall` although the max-length is 27 and we can keep going until we reach it but since our slug is going to be something like this `how-to-set-up-a-firewall-wi` we ignore the last word.

Reply:
Interesting topic - and there are plenty of resources that suggest one should even remove stop or filler words (like `the`, `a`, `an`, `of` and so on (source: https://stackoverflow.com/questions/2921037/what-is-the-ideal-length-of-an-url-slug). But I am no keen friend of that.

Wordpress seems to handle up to 200 characters for a slug.  What we could do is to check if the title is longer (after transforming it into a slug) and go back to the last "-" character and remove everything after (like your firewall example).

I checked my longest title on my blog - and that would be *"[Array, List, Collection, Set, ReadOnlyList - what? A comprehensive and exhaustive list of collection-like types](https://steven-giesel.com/blogPost/f11aadf5-5a55-4742-b046-28dddd0b3ebf)"*. In total: 112 characters. So if we want to have a safe margin something like 160 chars should be enough.

I am still thinking that we might want to persist this in the future. As you said, that is something somewhat immutable, so there might be no need to "calculate" this always on the fly.

Reply:
Agree we can persist this in the future.

Just a question why sometimes special char should removed but sometimes replaced with `-` 
For example 
`blog,title` -> `blog-title`
`blog?!title` -> `blogtitle`
I guess we should treat them in the same way, shouldn't we?
`blog,title` -> `blogtitle`
`blog?!title` -> `blogtitle`

Reply:
Fair - to be honest I am not sure how everything will persist in unicode form C and D. 
I am fine with your changes, as long as we have them "documentated" (in forms of tests).